L1Trigger/L1THGCal/interface/be_algorithms/HGCalTriggerCellCalibration.h

```diff
@@ -0,0 +1,40 @@
+#ifndef __L1Trigger_L1THGCal_HGCalTriggerCellCalibration_h__
+#define __L1Trigger_L1THGCal_HGCalTriggerCellCalibration_h__
+
+#include <stdint.h>
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCEEDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCHEDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+
+class HGCalTriggerCellCalibration{
+
+public:
+
+    HGCalTriggerCellCalibration(const edm::ParameterSet& conf);    
+ 
+    l1t::HGCalTriggerCell calibTrgCell(l1t::HGCalTriggerCell&, const edm::EventSetup& es); 
+    //void calibTrgCellCollection( l1t::HGCalTriggerCellBxCollection& trgCellColl, const edm::EventSetup& es);
+    void print();
+    edm::ESHandle<HGCalGeometry> hgceeGeoHandle_;
+    edm::ESHandle<HGCalGeometry> hgchefGeoHandle_;
+
+private:
+    
+    double LSB_;
+    uint32_t trgCellTruncBit_;
```

Comment:
Here the LSB that you give is not the LSB of the trigger cell energy that arrives in the backend. It is the LSB of the energy before the compression. 
What you should give instead is the LSB of the trigger cell energy after compression, which means that you don't need to pass the number of truncated bits. You just pass directly the correct LSB (which is computed from linLSB and the number of truncation bits).